src/kudu/consensus/raft_consensus.cc

```diff
@@ -3761,6 +3761,81 @@ Status RaftConsensus::GetVoterDistribution(std::map<std::string, int32> *vd) con
   return cmeta_->voter_distribution(vd);
 }
 
+Status RaftConsensus::ChangeUseQuorumId(bool val, bool force) {
+  TRACE_EVENT2("consensus", "RaftConsensus::ChangeTopologyConfig", "peer",
+               peer_uuid(), "tablet", options_.tablet_id);
+  ThreadRestrictions::AssertWaitAllowed();
+  LockGuard l(lock_);
+
+  if (!force) {
+    Status s = CheckNoConfigChangePendingUnlocked();
+    RETURN_NOT_OK(s);
+  }
+
+  RaftConfigPB config = cmeta_->ActiveConfig();
+
+  config.mutable_commit_rule()->set_use_quorum_id(val);
+
+  cmeta_->set_active_config(config);
+  CHECK_OK(cmeta_->Flush());
+
+  if (cmeta_->active_role() == RaftPeerPB::LEADER) {
+      RETURN_NOT_OK(RefreshConsensusQueueAndPeersUnlocked());
+  }
+  return Status::OK();
+}
+
+bool RaftConsensus::GetUseQuorumId() const {
+  ThreadRestrictions::AssertWaitAllowed();
+  LockGuard l(lock_);
+  return cmeta_->ActiveConfig().commit_rule().use_quorum_id();
+}
+
+Status RaftConsensus::SetPeerQuorumIds(std::map<std::string, std::string> uuid2quorum_ids,
+                                       bool force) {
+  ThreadRestrictions::AssertWaitAllowed();
+  LockGuard l(lock_);
+
+  if (!force && cmeta_->ActiveConfig().commit_rule().use_quorum_id()) {
+    return Status::Aborted("Not allow change quorum_id when it is being"
+                           "used, unless force.");
```

Comment:
Should we ever force this? If we need to force it feels like we should force it to turn off first then change the ids.

Reply:
I think it will be very powerful but dangerous at the same time. Potentially it can be used by unsafe_config_change, for example, let all healthy instances construct a healthy quorum to unblock certain situation.

Reply:
"Peer quorum ids cannot be changed when in use unless forced."